schema/loader.go

```diff
@@ -0,0 +1,165 @@
+	PresentationSubmissionSchema              SchemaFile = "pe-presentation-submission.json"
+	PresentationClaimFormatDesignationsSchema SchemaFile = "pe-definition-claim-format-designations.json"
+	SubmissionClaimFormatDesignationsSchema   SchemaFile = "pe-submission-claim-format-designations.json"
+	SubmissionRequirementSchema               SchemaFile = "pe-submission-requirement.json"
+	SubmissionRequirementsSchema              SchemaFile = "pe-submission-requirements.json"
+
+	PresentationClaimFormatDesignationFormatDefinition SchemaFile = "pe-submission-claim-format-designations.json"
+
+	// Credential Manifest Schemas
+	CredentialManifestSchema    SchemaFile = "cm-credential-manifest.json"
+	CredentialApplicationSchema SchemaFile = "cm-credential-application.json"
+	CredentialResponseSchema    SchemaFile = "cm-credential-response.json"
+	OutputDescriptorsSchema     SchemaFile = "cm-output-descriptors.json"
+
+	// Wallet Rendering Schemas
+	DisplayMappingObjectSchema        SchemaFile = "wr-display-mapping-object.json"
+	EntityStylesSchema                SchemaFile = "wr-entity-styles.json"
+	LabeledDisplayMappingObjectSchema SchemaFile = "wr-labeled-display-mapping-object.json"
+
+	// VC JSON Schema Schemas
+	VerifiableCredentialJSONSchemaSchema SchemaFile = "vc-json-schema.json"
+)
+
+func (s SchemaFile) String() string {
+	return string(s)
+}
+
+// CachingLoader is a struct that holds local schemas
+type CachingLoader struct {
+	schemas map[string]string
+}
+
+// NewCachingLoader returns a new CachingLoader that enables the ability to cache http and https schemas
+func NewCachingLoader(schemas map[string]string) (*CachingLoader, error) {
+	cl := CachingLoader{schemas: make(map[string]string)}
+	for schemaURI, schema := range schemas {
+		if _, ok := cl.schemas[schemaURI]; ok {
+			return nil, fmt.Errorf("schema %q already exists", schemaURI)
+		}
+		cl.schemas[schemaURI] = schema
+	}
+	return &cl, nil
+}
+
+// EnableHTTPCache enables caching of http and https schemas
+func (cl *CachingLoader) EnableHTTPCache() {
+	jsonschema.Loaders["http"] = cl.cachingLoaderForProtocol("http")
+	jsonschema.Loaders["https"] = cl.cachingLoaderForProtocol("https")
+}
+
+func (cl *CachingLoader) cachingLoaderForProtocol(protocol string) func(url string) (io.ReadCloser, error) {
+	var m sync.Mutex
+	return func(url string) (io.ReadCloser, error) {
+		// make sure only one process can write to the map at a time
+		m.Lock()
+		defer m.Unlock()
+
+		// see if the schema is in the cache already
+		schema, ok := cl.schemas[strings.TrimPrefix(url, protocol+"://")]
+		if ok {
+			return io.NopCloser(strings.NewReader(schema)), nil
+		}
+
+		// fallback lookup if it's stored with the fully qualified url
+		schema, ok = cl.schemas[url]
+		if ok {
+			return io.NopCloser(strings.NewReader(schema)), nil
+		}
+
+		// load from the internet
+		loaded, err := httploader.Load(url)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to load schema from %s", protocol)
```

Comment:
So that the error reads better
```suggestion
			return nil, errors.Wrapf(err, "loading schema from %s", protocol)
```